starport/pkg/protoanalysis/import.go

```diff
@@ -0,0 +1,44 @@
+package protoanalysis
+
+import (
+	"fmt"
+
+	"github.com/tendermint/starport/starport/pkg/placeholder"
+)
+
+// IsProtoImported returns true if the proto file is imported in the provided proto file
+func IsProtoImported(path, protoImport string) (bool, error) {
+	f, err := ParseFile(path)
+	if err != nil {
+		return false, err
+	}
+
+	protoImport = fmt.Sprintf("%s.proto", protoImport)
+	for _, dep := range f.Dependencies {
+		if dep == protoImport {
+			return true, nil
+		}
+	}
+
+	return false, nil
+}
+
+// EnsureProtoImported add the proto file import in the proto file content in case it's not defined
+func EnsureProtoImported(content, protoImport, protoPath, importPlaceholder string, replacer placeholder.Replacer) string {
+	isImported, err := IsProtoImported(protoPath, protoImport)
+	if err != nil {
+		replacer.AppendMiscError(fmt.Sprintf("failed to check %s dependency %s", protoImport, err.Error()))
+		return content
+	}
+	if !isImported {
+		templateGogoProtoImport := `%[1]v
```

Comment:
we cannot let protoanalysis be aware of templates. also, protoanalysis is only responsible to analyse existent proto files, it should not mutate or show how to mutate them.

Reply:
I removed the replacer pkg import from protoanalysis. Now the method only returns the strings of the new proto import, and the caller needs to use the replaces if want. I think it's good now... Any thoughts on this?

Reply:
can I see which lines/commits are we talking about?